src/utils/ts/FocusManager.tsx

```diff
@@ -48,7 +49,11 @@ const FocusManager: React.FC<FocusManagerProps> = ({ children, refs = undefined,
       refs[focusedIndex].focus();
     };
 
-    updateFocused();
+    if (prevFocusedIndex.current !== focusedIndex) {
+      updateFocused();
+    }
+
+    prevFocusedIndex.current = focusedIndex;
```

Comment:
We could move this assignment into the `if` to improve semantics.

Reply:
If we move the assignment inside the if we will not set the `prevFocusedIndex.current` on mount though, right?

Reply:
That would happen on line 22, I think (the assignment on mount-time).

Reply:
Yes you're right, I forgot that we're initializing the ref when we're declaring it. I will move it to the `if`.